tests/rptest/remote_scripts/w_workload.py

```diff
@@ -0,0 +1,200 @@
+    def __init__(self, bootstrap, topic):
+        self.bootstrap = bootstrap
+        self.topic = topic
+
+    def check(self, history, timeout_s):
+        for partition in history.keys():
+            logging.debug(f"validating {self.topic}/{partition}")
+            begin = time.time()
+            consumer = Consumer({
+                "bootstrap.servers": self.bootstrap,
+                "group.id": "group-kaf",
+                "enable.auto.commit": False,
+                "auto.offset.reset": "earliest",
+                "isolation.level": "read_committed"
+            })
+            consumer.assign(
+                [TopicPartition(self.topic, partition, OFFSET_BEGINNING)])
+
+            i = 0
+            while True:
+                if i == len(history[partition]):
+                    break
+                if time.time() - begin > timeout_s:
+                    raise Exception("timeout")
+                msgs = consumer.consume(timeout=10)
+                for msg in msgs:
+                    if msg.error() != None:
+                        logging.error(msg.error())
+                    offset = msg.offset()
+                    key = msg.key().decode('utf-8')
+                    value = msg.value().decode('utf-8')
+                    written = history[partition][i]
+                    if offset != written[0]:
+                        raise Exception("mismatched offset")
+                    if key != written[1]:
+                        raise Exception("mismatched key")
+                    if value != written[2]:
+                        raise Exception("mismatched value")
+                    i += 1
+
+
+class ProducingThread:
+    def __init__(self):
+        self.is_active = False
+        self.is_broken = False
+        self.thread = None
+        self.topic = None
+        self.count = -1
+        self.connection = None
+        self.keep_history = False
+        self.history = dict()
+
+
+class Service:
+    def __init__(self):
+        self.threads = dict()
+
+    def validate(self, name, timeout_s):
+        if name not in self.threads:
+            raise Exception(f"unknown name {name}")
+        thread = self.threads[name]
+        if thread.is_broken:
+            raise Exception("broken")
+        if thread.is_active:
+            raise Exception("active")
+        if not (thread.keep_history):
+            raise Exception("w/o history")
+        checker = LogChecker(thread.connection, thread.topic)
+        checker.check(thread.history, timeout_s)
+
+    def process(self, thread):
+        try:
+            client = SyncProducer(thread.connection)
+            client.init()
+
+            for i in range(0, thread.count):
+                key = f"key-{str(i).zfill(8)}"
+                value = f"record-{str(i).zfill(8)}"
+                info = client.produce(thread.topic, key.encode('utf-8'),
+                                      value.encode('utf-8'))
+                if thread.keep_history:
```

Comment:
There's an implicit memory limit on how many messages this can produce while keeping an array of history -- maybe it's worth explicitly putting a maximum into this class to avoid tripping up someone who tries to use it for very high message counts in future?

Or, ideally, we would do verification continuously so that we didn't have to keep an unbounded size array.

Reply:
I've just converted an already existing test which also didn't have any limits. By the way the java generator / checker is unbounded, let me check if I may just use it there and abandon this one.

Reply:
I've switched to unbounded java checker